go/state/state_db.go

```diff
@@ -827,6 +828,33 @@ func (s *stateDB) SetState(addr common.Address, key common.Key, value common.Val
 	}
 }
 
+func (s *stateDB) HasEmptyStorage(address common.Address) bool {
+	var (
+		isEmpty, shouldReturn bool
+	)
+	if !s.Exist(address) {
```

Comment:
The joke is: an account may not exist but have some non-empty storage. This is what EIP-7610 is addressing. So I think  the `Exist` check should not be done her.

Reply:
I agree, after your explanation above, this check makes zero sense